pyuvdata/utils.py

```diff
@@ -388,6 +388,69 @@ def unphase_uvw(ra, dec, uvw):
     return(xyz)
 
 
+def uvcalibrate(uvdata, uvcal, inplace=True, prop_flags=True, flag_missing=True):
+    """
+    Calibrate a UVData object with a UVCal object.
+
+    Parameters
+    ----------
+    uvdata: UVData object
+    uvcal: UVCal object
+    inplace: bool, optional
+        if True edit uvdata in place, else deepcopy
+    prop_flags : bool, optional
+        if True, propagate calibration flags to data flags
+        and doesn't use flagged gains. Otherwise, uses flagged gains and
+        does not propagate calibration flags to data flags.
+    flag_missing: bool, optional
+        if True, flag baselines in uvdata
+        if a participating antenna or polarization is missing in uvcal.
+
+    Returns
+    -------
+    UVData
+        Retruns if not inplace
+    """
+    # deepcopy for not inplace
+    if not inplace:
+        uvdata = copy.deepcopy(uvdata)
+
+    # input checks
+    if uvcal.cal_type != 'gain':
+        raise ValueError("uvcal must have cal_type of 'gain' for calibration")
```

Comment:
As a user, if I had a delay type UVCal object, I would assume the calibrate function would convert to gain and apply (even if it made a copy of the UVCal object to not disturb the original).